rs/macros/core/src/shared.rs

```diff
@@ -223,3 +223,48 @@ fn replace_lifetime_with_static_in_path_args(path_args: PathArguments) -> PathAr
         path_args
     }
 }
+
+/// Check if type is `CommandReply<T>` and extract inner type `T`
+pub(crate) fn extract_reply_type_with_value(ty: &Type) -> Option<Type> {
+    match ty {
+        Type::Path(tp) => extract_reply_result_type(tp),
+        Type::ImplTrait(imp) => extract_reply_result_type_from_impl_into(imp),
+        _ => None,
+    }
+}
+
+/// Extract `T` type from `CommandReply<T>`
+fn extract_reply_result_type(tp: &TypePath) -> Option<Type> {
+    if let Some(last) = tp.path.segments.last() {
+        if last.ident != "CommandReply" {
+            return None;
+        }
+        if let PathArguments::AngleBracketed(args) = &last.arguments {
+            if args.args.len() == 1 {
+                if let Some(GenericArgument::Type(ty)) = args.args.first() {
+                    return Some(ty.clone());
+                }
+            }
+        }
+    }
+    None
+}
+
+/// Extract `T` type from `impl Into<CommandReply<T>>`
```

Comment:
I don't like much cryptography in code. becomes unreadable + adds up to maintenance